doc/plugins.dox

```diff
@@ -444,30 +535,170 @@ native spelling inside parenthesis, where applicable.
+widget of that dialog to the combined plugin dialog simply by returning it from @ref
+GeanyPluginFuncs::configure. You don't actually add it, Geany will do that for you. The pointer
+to the dialog is passed to @a configure simply to allow you to connect to its "response" or "close"
+signals.
+
+The following lists the function mapping of previous @a plugin_* functions to the new @a
+GeanyPlugin::funcs. They are semantically the same, however the new functions receive more
+parameters which you may use or not.
+
+  - plugin_init() => GeanyPlugin->funcs->init
+  - plugin_cleanup() => GeanyPlugin->funcs->cleanup
+  - plugin_help() => GeanyPlugin->funcs->help
+  - plugin_configure() => GeanyPlugin->funcs->configure
+
+@note @ref GeanyPluginFuncs::init() should return a boolean value: whether or not the plugin loaded
+succesfully. Since legacy plugins couldn't fail in plugin_init() you should return @c TRUE
+unconditionally.
+
+@note Again, plugin_configure_single() is not supported anymore.
+
+@subsection Variables
+
+Exported global variables are not recognized anymore. They are replaced in the following ways:
+
+@ref plugin_info is simply removed. Instead, you have to assign the values to GeanyPlugin::info
+yourself, and it must be done inside your @a geany_load_module().
+
+Example:
+
+@code
+PLUGIN_SET_INFO(
+	"HelloWorld",
+	"Just another tool to say hello world",
+	"1.0", "John Doe <john.doe@example.org>");
+@endcode
+
+becomes
+
+@code
+G_MODULE_EXPORT
+void geany_load_module(GeanyPlugin *plugin, GModule *module, gint geany_api_version)
+{
+// ...
+	plugin->info->name = "HelloWorld";
+	plugin->info->description = "Just another tool to say hello world";
+	plugin->info->version = "1.0";
+	plugin->info->author = "John Doe <john.doe@example.org>";
+// ...
+}
+@endcode
+@note Refer to @ref translatable_plugin_information for i18n support for the metadata.
+
+
+The @ref plugin_callbacks array is supported by assigning the @ref GeanyPluginFuncs::callbacks to
+the array.
+
+@ref plugin_fields is not supported anymore. Use ui_add_document_sensitive() instead. @a
+plugin_key_group_info and @ref plugin_key_group are also not supported anymore. Use
+plugin_set_key_group() and keybindings_set_item() instead.
+
+Additionally, Geany tradionionally set a few variables. This is not the case anymore. @ref
+geany_functions has been removed in 1.25 and since then existed only for compatibility and has been
+empty. You can simply remove its declaration from your source code. @ref geany_plugin is passed to
+each @ref GeanyPluginFuncs function. You need to store it yourself somewhere if you need it elsewhere. @ref
+geany_data is now available as a member of GeanyPlugin.
+
+@code
+GeanyPlugin *geany_plugin;
+GeanyData *geany_data;
+
+static gboolean my_init(GeanyPlugin *plugin, gpointer pdata)
+{
+// ...
+	geany_plugin = plugin;
+	geany_data = plugin->geany_data;
+	return TRUE;
+}
+@endcode
```

Comment:
Am 12. August 2015 17:14:52 MESZ, schrieb Colomban Wendling <notifications@github.com>:
>> +empty. You can simply remove its declaration from your source code.
>@ref geany_plugin is passed to
>> +each @ref GeanyPluginFuncs function. You need to store it yourself
>somewhere if you need it elsewhere. @ref
>> +geany_data is now available as a member of GeanyPlugin.
>> +
>> +@code
>> +GeanyPlugin *geany_plugin;
>> +GeanyData *geany_data;
>> +
>> +static gboolean my_init(GeanyPlugin *plugin, gpointer pdata)
>> +{
>> +// ...
>> +	geany_plugin = plugin;
>> +	geany_data = plugin->geany_data;
>> +	return TRUE;
>> +}
>> +@endcode
>
>Hum, I don't think I'd advertize with code doing such a minimalistic
>port.  It would work, sure, but wouldn't really give much of the
>cleanup benefits of the new API, and a clever person would figure she
>could do this anyway.
>
>Or at least also suggest to rather store this in some dynamic state or
>something maybe?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/geany/geany/pull/469/files#r36871816

This is about porting legacy plugins with the least possible effort. This should be advertised because it increases the chance that more plugins are ported earlier. Advertising a more involved porting effort will raise the bar and we have to wait longer until we can remove the legacy support.

I prefer to keep it this way. It's not like it's buggy or anything, just doesn't match your fine taste :-) but it's not about core code but third party plugins